lib/matplotlib/legend.py

```diff
@@ -1260,6 +1260,8 @@ def _parse_legend_args(axs, *args, handles=None, labels=None, **kwargs):
     elif len(args) >= 2:
         handles, labels = args[:2]
         extra_args = args[2:]
+    elif len(handles)!= len(labels):
```

Comment:
This condition will never be evaluated because the one of the other conditionals in this if/elif chain will always be `True` first.

Reply:
Yes I see that now. 
An entirely new block would do the trick or we should rethink like you said. This issue was tagged as good first issue so and solution seemed pretty straightforward at the time.will analyslze along your directives

Reply:
I think an entirely new block should be enough. This conditional statement (L1222:L1262) is to assign values to `labels` and `handles`. Checking the validity of `handles` and `labels` should come in a new conditional statement after that.